### DISCUSSION TITLE
[mmclassification 1.X]How can I record the testing inference results of each images in each category

### DISCUSSION THREAD

**Discussion by ThomaswellY** [open-mmlab/mmclassification#1444] [Q&A]
following the mmclassification 1.X tutor manual，the testing inference is successful, and the record on log is shown as follow: (max_epoch=1, logger_inter=100)
{"lr": 0.1, "data_time": 0.0008228302001953125, "loss": 1.4076386213302612, "time": 0.08029277324676513, "epoch": 1, "memory": 2455, "step": 100}
{"lr": 0.1, "data_time": 0.0007920742034912109, "loss": 1.3337437868118287, "time": 0.07977149486541749, "epoch": 1, "memory": 2455, "step": 200}
{"lr": 0.1, "data_time": 0.0008879423141479493, "loss": 1.3094250798225402, "time": 0.08056514263153076, "epoch": 1, "memory": 2455, "step": 300}
{"single-label/precision": 42.19956970214844, "single-label/recall": 35.5002326965332, "single-label/f1-score": 29.484556198120117, "accuracy/top1": 46.23514175415039, "data_time": 0.00510249137878418, "time": 0.02877798080444336, "step": 1}
However, the final evaluation of testing inferece is over the whole testing dataset, which is not enough for me.
I have consult "mmengine.readthedoc.io" for help, and seems mmengine.hook or mmengine.runner doesn't support such functional requirements. 
by the way , in branch of mmclassification 0.X,  the whole testing inference result is in form of vector , and can be ultilized easily from "outputs = single_gpu_test(model, data_loader, args.show, args.show_dir,**show_kwargs)"in tools/test.py. while in 1.X branch, the corn is hidden deeper.
Sincerely looking forward to your response, Thanks in advance~

**Comment by ThomaswellY** [comment]
@Ezra-Yu @mzr1996

**Comment by Ezra-Yu** [comment]
Use `--out-item pred` refer to https://github.com/open-mmlab/mmclassification/blob/c21d99ed3ef881bb17c0cc76590bfc4e99169b95/tools/test.py#L23

**Reply by ThomaswellY** [reply]
thanks for you apply, maybe my explanation is not clear. --out-item is used in tools/test.py in master branch. I wonder how to achieve the same function in 1.X branch.

**Reply by Ezra-Yu** [reply]
it is in the dev-1.x branch. maybe you need to update the code.

**Reply by ThomaswellY** [reply]
Thanks my friend, I will try it soon. and do you mean that the 1.X branch is for you developer and is still under developing, and dev-1.X branch is for we users, and any stable modification will merge from 1.X into dev-1.X?  
I am a little confused, which branch is the stable one?

**Reply by ThomaswellY** [reply]
Maybe I'd better just wait for dev-1.X merge into 1.X.  "--out-item" is just added in dev-1.X on Feb this year, seems still long time for its' next time to merge into 1.X.
